include/c2pa.h

```diff
@@ -135,15 +213,284 @@ char *c2pa_sign_file(const char *source_path,
                      const char *data_dir);
 
 /**
- * Releases a string allocated by Rust
+ * Frees a string allocated by Rust
  *
  * # Safety
  * Reads from null terminated C strings
  * The string must not have been modified in C
- * can only be released once and is invalid after this call
+ * can only be freed once and is invalid after this call
+ */
+IMPORT extern void c2pa_string_free(char *s);
+
+/**
+ * Frees a string allocated by Rust
+ *
+ * Provided for backward api compatibility
+ * # Safety
+ * Reads from null terminated C strings
+ * The string must not have been modified in C
+ * can only be freed once and is invalid after this call
  */
 IMPORT extern void c2pa_release_string(char *s);
 
+/**
+ * Creates a C2paReader from an asset stream with the given format
+ * # Errors
+ * Returns NULL if there were errors, otherwise returns a pointer to a ManifestStore
+ * The error string can be retrieved by calling c2pa_error
+ * # Safety
+ * Reads from null terminated C strings
+ * The returned value MUST be released by calling c2pa_reader_free
+ * and it is no longer valid after that call.
+ * # Example
+ * ```c
+ * auto result = c2pa_reader_from_stream("image/jpeg", stream);
+ * if (result == NULL) {
+ *   printf("Error: %s\n", c2pa_error());
+ * }
+ * ```
+ */
+IMPORT extern
+struct C2paReader *c2pa_reader_from_stream(const char *format,
+                                           struct CStream *stream);
+
+/**
+ * Frees a C2paReader allocated by Rust
+ * # Safety
+ * can only be freed once and is invalid after this call
+ */
+IMPORT extern void c2pa_reader_free(struct C2paReader *reader_ptr);
+
+/**
+ * Returns a JSON string generated from a C2paReader
+ *
+ * # Safety
+ * The returned value MUST be released by calling c2pa_string_free
+ * and it is no longer valid after that call.
+ */
+IMPORT extern char *c2pa_reader_json(struct C2paReader *reader_ptr);
+
+/**
+ * writes a C2paReader resource to a stream given a uri
+ * # Errors
```

Comment:
Style consistency question: In this function you use a blank line prior to each of the `#` headings; in most others you don't. I prefer this format, but even more, I vote for consistency.

Reply:
Looks like this is still an issue.